src/Command/DrupalUpdate/UpdateScriptUtility.php

```diff
@@ -0,0 +1,317 @@
+      // Replace the folder to inside docroot folder.
+      $this->coreUpdate($save_to . '/drupal');
+      $this->rrmdir($save_to);
+    }
+  }
+
+  /**
+   * Remove directory and sub directory.
+   * @param $dir
+   */
+  function rrmdir($dir) {
+    if (is_dir($dir)) {
+      $objects = scandir($dir);
+      foreach ($objects as $object) {
+        if ($object != "." && $object != "..") {
+          if (filetype($dir . "/" . $object) == "dir")
+                        $this->rrmdir($dir . "/" . $object);
+          else unlink   ($dir . "/" . $object);
+        }
+      }
+      reset($objects);
+      rmdir($dir);
+    }
+  }
+
+  /**
+   * After extraction copy code temp to main folder replace.
+   * Core modules, themes, profile.
+   * @param $actual_di_path
+   * @param $replace_dir_path
+   */
+  function coreUpdate($actual_di_path) {
+    $dir    = $actual_di_path;
+    $replace_dir_path = str_replace('/temp_drupal_core/drupal', '', $actual_di_path);
+    $files1 = array_diff(scandir($dir), ['.', '..']);
+
+    foreach($files1 as $r_dir => $c_dir){
+      $tm_path = $dir . "/" . $c_dir;
+      if(!in_array($c_dir, $this->ignore_updates_files)){
+        if(is_dir($tm_path)){
+          $this->custom_copy($dir . '/' . $c_dir, $replace_dir_path . '/' . $c_dir);
+          continue;
+        }
+        rename($dir . '/' . $c_dir, $replace_dir_path . '/' . $c_dir);
+      }
+    }
+  }
+
+  /**
+   * Copy folders and subfolder from temp to main folder.
+   * @param $src
+   * @param $dst
+   */
+  function custom_copy($src, $dst) {
+
+    // open the source directory
+    $dir = opendir($src);
+
+    // Make the destination directory if not exist
+    @mkdir($dst);
+
+    // Loop through the files in source directory
+    while( $file = readdir($dir) ) {
+
+      if (( $file != '.' ) && ( $file != '..' )) {
+        if ( is_dir($src . '/' . $file) )
+                {
+
+          // Recursively calling custom copy function
+          // for sub directory
+          $this->custom_copy($src . '/' . $file, $dst . '/' . $file);
+
+        }
+        else {
+          copy($src . '/' . $file, $dst . '/' . $file);
+        }
+      }
+    }
+
+    closedir($dir);
+  }
+
+  /**
+   * Remove after copy tar files and temp folder.
+   * @param $remove_file_list
+   */
+  function unlinktarfiles($remove_file_list) {
+
+    foreach ($remove_file_list as $k => $value){
+      if( ($k == 0) || ($value['package_type']=='core') ){
+        continue;
+      }
+      //echo $value['file_path']."/".$value['package'].".tar.gz \n";
+      if(is_array($value['file_path'])){
+        foreach ($value['file_path'] as $item){
+          unlink($item . "/" . $value['package'] . ".tar.gz");
+        }
+      }else{
+        unlink($value['file_path'] . "/" . $value['package'] . ".tar.gz");
+      }
+    }
+  }
+
+  /**
+   * Compare two versions.
+   * @param $v1
+   * @param $v2
+   * @return int
+   */
+  function versionCompare($v1, $v2) {
```

Comment:
PHP has version_compare function (https://www.php.net/manual/en/function.version-compare.php). See if you can use that.

Reply:
Or, you can use https://packagist.org/packages/composer/semver